scripts/setup-mapbox-sdk.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Mapbox SDK Configuration Script for iOS and Android
+ * ===================================================
+ *
+ * Purpose:
+ * --------
+ * This script configures the development environment to download Mapbox SDKs
+ * for both iOS and Android builds. We use Mapbox to display maps in the App. As Mapbox SDKs
+ * are closed-sourced, we need to authenticate with Mapbox during the download.
+ *
+ * Background:
+ * -----------
+ * To build the App, a machine requires a Mapbox token with the necessary permissions to download SDKs.
+ * This token needs to be saved in relevant files, enabling tools like CocoaPods for iOS or Gradle
+ * for Android to access the Mapbox SDK during the build process.
+ *
+ * The `.netrc` file for iOS Configuration:
+ * ----------------------------------------
+ * The token for iOS is stored in the `.netrc` file located in the user's home directory.
+ * This file is used in Unix-like systems to store credentials for remote machine access.
+ *
+ * The `gradle.properties` file for Android Configuration:
+ * -------------------------------------------------------
+ * The token for Android is stored in the `gradle.properties` file located in the .gradle directory
+ * in the user's home. This is accessed by the Android build system during the SDK download.
+ *
+ * Usage:
+ * ------
+ * To run this script, pass the secret Mapbox access token as a command-line argument:
+ * `ts-node ./scripts/setup-mapbox-sdk.ts YOUR_MAPBOX_ACCESS_TOKEN`
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+const NETRC_PATH = path.join(process.env.HOME || '', '.netrc');
+const USER_GRADLE_DIR = path.join(process.env.HOME || '', '.gradle');
+const GRADLE_PROPERTIES_PATH = path.join(USER_GRADLE_DIR, 'gradle.properties');
+
+export async function saveMapboxToken(token: string) {
+    try {
+        // iOS Configuration for .netrc
+        console.log(`Configuring ${NETRC_PATH} for Mapbox iOS SDK download`);
+        if (fs.existsSync(NETRC_PATH) && fs.readFileSync(NETRC_PATH, 'utf8').includes('api.mapbox.com')) {
+            const updatedContent = fs.readFileSync(NETRC_PATH, 'utf8').replace(/password .*/, `password ${token}`);
+            fs.writeFileSync(NETRC_PATH, updatedContent, 'utf8');
+            console.log(`Token was updated in ${NETRC_PATH}`);
+        } else {
+            fs.appendFileSync(NETRC_PATH, `machine api.mapbox.com\nlogin mapbox\npassword ${token}\n`, 'utf8');
+            console.log(`${NETRC_PATH} was configured with new credentials`);
+        }
+
+        // Set the permissions of the .netrc file to ensure it's kept private
+        await fs.chmodSync(NETRC_PATH, 0o600);
```

Comment:
If you're using `chmodSync()` you don't need `await` anyways, right? It's already synchronous.